src/libAtomVM/mailbox.c

```diff
@@ -132,75 +136,75 @@ void mailbox_send(Context *c, term t)
+    msg->msg_memory_size = estimated_mem_usage;
 
-    mailbox_post_message(c, m);
+    mailbox_post_message(c, &msg->base);
 }
 
 void mailbox_send_term_signal(Context *c, enum MessageType type, term t)
 {
     unsigned long estimated_mem_usage = memory_estimate_usage(t);
 
-    MailboxMessage *m = malloc(sizeof(struct MessageHeader) + sizeof(struct TermSignal) + estimated_mem_usage * sizeof(term));
-    if (IS_NULL_PTR(m)) {
+    struct TermSignal *ts = malloc(sizeof(struct TermSignal) + estimated_mem_usage * sizeof(term));
+    if (IS_NULL_PTR(ts)) {
         fprintf(stderr, "Failed to allocate memory: %s:%i.\n", __FILE__, __LINE__);
         return;
     }
-    m->body.term.mso_list = term_nil();
+    ts->mso_list = term_nil();
 
-    term *heap_pos = mailbox_message_memory(&m->body.term.signal_term);
-    m->header.type = type;
-    m->body.term.signal_term = memory_copy_term_tree(&heap_pos, t, &m->body.term.mso_list);
-    m->body.term.msg_memory_size = estimated_mem_usage;
+    term *heap_pos = mailbox_message_memory(&ts->signal_term);
+    ts->base.type = type;
+    ts->signal_term = memory_copy_term_tree(&heap_pos, t, &ts->mso_list);
+    ts->msg_memory_size = estimated_mem_usage;
 
-    mailbox_post_message(c, m);
+    mailbox_post_message(c, &ts->base);
 }
 
 void mailbox_send_built_in_atom_signal(Context *c, enum MessageType type, term atom)
 {
-    MailboxMessage *m = malloc(sizeof(struct MessageHeader) + sizeof(struct BuiltInAtomSignal));
-    if (IS_NULL_PTR(m)) {
+    struct BuiltInAtomSignal *atom_signal = malloc(sizeof(struct BuiltInAtomSignal));
+    if (IS_NULL_PTR(atom_signal)) {
         fprintf(stderr, "Failed to allocate memory: %s:%i.\n", __FILE__, __LINE__);
         return;
     }
-    m->header.type = type;
-    m->body.atom.atom = atom;
+    atom_signal->base.type = type;
+    atom_signal->atom = atom;
 
-    mailbox_post_message(c, m);
+    mailbox_post_message(c, &atom_signal->base);
 }
 
 void mailbox_send_built_in_atom_request_signal(Context *c, enum MessageType type, int32_t pid, term atom)
 {
-    MailboxMessage *m = malloc(sizeof(struct MessageHeader) + sizeof(struct BuiltInAtomRequestSignal));
-    if (IS_NULL_PTR(m)) {
+    struct BuiltInAtomRequestSignal *atom_request = malloc(sizeof(struct BuiltInAtomRequestSignal));
+    if (IS_NULL_PTR(atom_request)) {
         fprintf(stderr, "Failed to allocate memory: %s:%i.\n", __FILE__, __LINE__);
         return;
     }
-    m->header.type = type;
-    m->body.atom_request.sender_pid = pid;
-    m->body.atom_request.atom = atom;
+    atom_request->base.type = type;
+    atom_request->sender_pid = pid;
+    atom_request->atom = atom;
 
-    mailbox_post_message(c, m);
+    mailbox_post_message(c, &atom_request->base);
 }
 
 void mailbox_send_empty_body_signal(Context *c, enum MessageType type)
 {
-    MailboxMessage *m = malloc(sizeof(struct MessageHeader));
+    MailboxMessage *m = malloc(sizeof(struct MailboxMessage));
```

Comment:
We still have a struct and a typedef here.

Reply:
Fixed.